src/public/services/__tests__/AuthService.test.ts

```diff
@@ -0,0 +1,60 @@
+import axios from 'axios'
+import { mocked } from 'ts-jest/utils'
+import { Opaque } from 'type-fest'
+
+import {
+  AUTH_ENDPOINT,
+  checkIsEmailAllowed,
+  sendLoginOtp,
+} from '../AuthService'
+
+jest.mock('axios')
+
+const MockAxios = mocked(axios, true)
+
+// Duplicated here instead of exporting from AuthService to prevent production
+// code from casting to Email type without going through a type guard/validator.
+type TestEmail = Opaque<string, 'Email'>
+
+describe('AuthService', () => {
+  describe('checkIsEmailAllowed', () => {
+    const EXPECTED_POST_ENDPOINT = `${AUTH_ENDPOINT}/email/validate`
+
+    it('should return given email argument when email is allowed', async () => {
```

Comment:
how about a test case when email domain is not whitelisted?

Reply:
that test case is already tested on the server, these tests are just testing that the endpoint is called